gtsam/sfm/DsfTrackGenerator.cpp

```diff
@@ -0,0 +1,136 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file DsfTrackGenerator.cpp
+ * @date October 2022
+ * @author John Lambert
+ * @brief Identifies connected components in the keypoint matches graph.
+ */
+
+#include <gtsam/sfm/DsfTrackGenerator.h>
+
+#include <algorithm>
+#include <iostream>
+
+namespace gtsam {
+
+namespace gtsfm {
+
+typedef DSFMap<IndexPair> DSFMapIndexPair;
+
+/// Generate the DSF to form tracks.
+static DSFMapIndexPair generateDSF(const MatchIndicesMap& matches) {
```

Comment:
is there a reason for these functions to be static?

Reply:
Yes. They are internal to a cpp file; it is customary to make them static.